piptools/resolver.py

```diff
@@ -87,14 +88,17 @@ def resolve(self, max_rounds=10):
 
     def _check_constraints(self):
         for constraint in chain(self.our_constraints, self.their_constraints):
-            if constraint.link is not None and not constraint.editable:
```

Comment:
This also removes the `not constraint.editable` condition — was this intentional? After trying it out, whether or not `-e` is added to VCS URLs seems to make no difference, but I don't know if it should.